### DISCUSSION TITLE
list - Incompatible types in assignment

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [python/typing#1205] [Q&A]
Have a list with different types ie.

```
mylist: list[Union[list[int], list[str], int]]
mylist = [[2, 4, 6, 8], ['1', '3'], 2, [10, 12]]
d0, d1, d2, d3 = mylist[0], mylist[1], mylist[2], mylist[3]
```
which gives the errors:
```
(d0) error: Incompatible types in assignment (expression has type "Union[List[int], List[str], int]", variable has type "List[int]")
(d1) error: Incompatible types in assignment (expression has type "Union[List[int], List[str], int]", variable has type "List[str]")
(d2) error: Incompatible types in assignment (expression has type "Union[List[int], List[str], int]", variable has type "int")
(d3) error: Incompatible types in assignment (expression has type "Union[List[int], List[str], int]", variable has type "List[int]")
```
Cheat Sheet says to use Union when something (mylist) could be one of a few types (list[int], list[str], int). What gives?

**Comment by erictraut** [comment]
Which type checker (and which version) are you using? Your sample type checks without errors in pyright and the latest published version of mypy.

**Comment by dineshbvadhia** [comment]
mypy 0.960
python 3.9.12
Windows 10

**Reply by erictraut** [reply]
I wonder if this is a mypy cache issue. Perhaps try deleting the cache and see if you can still repro?

**Comment by dineshbvadhia** [comment]
Yes, it was the mypy_cache.

Is there a way to flush to cache automatically ?

**Reply by erictraut** [reply]
Mypy does support a flag "--no-incremental" that tells it to ignore the cache when you run it, but performance is impacted pretty significantly when you do this.
